gotham/src/middleware/session/mod.rs

```diff
@@ -91,36 +94,74 @@ impl Default for SessionCookieConfig {
 impl SessionCookieConfig {
     fn to_cookie_string(&self, value: &str) -> String {
         // Ensure this is always enough to prevent re-allocs
+        let cookie_config = self.validate_prefix();
         let mut cookie_value = String::with_capacity(255);
 
-        cookie_value.push_str(&self.name);
+        cookie_value.push_str(&cookie_config.name);
         cookie_value.push('=');
         cookie_value.push_str(value);
 
-        if self.secure {
+        if cookie_config.secure {
             cookie_value.push_str("; Secure")
         }
 
-        if self.http_only {
+        if cookie_config.http_only {
             cookie_value.push_str("; HttpOnly")
         }
 
-        match self.same_site {
+        match cookie_config.same_site {
             SameSiteEnforcement::Strict => cookie_value.push_str("; SameSite=Strict"),
             SameSiteEnforcement::Lax => cookie_value.push_str("; SameSite=Lax"),
             SameSiteEnforcement::Disabled => (),
         }
 
-        if let Some(ref domain) = self.domain {
+        if let Some(ref domain) = cookie_config.domain {
             cookie_value.push_str("; Domain=");
             cookie_value.push_str(domain);
         }
 
         cookie_value.push_str("; Path=");
-        cookie_value.push_str(&self.path);
+        cookie_value.push_str(&cookie_config.path);
 
         cookie_value
     }
+
+    fn validate_prefix(&self) -> SessionCookieConfig {
```

Comment:
I realised this could return an `Option<SessionCookieConfig>` instead, so that in the case that it doesn't change it doesn't need to be copied. Is that a worthwhile change?

Reply:
(Avoiding the need to `.clone()` but also avoiding the need for an `Option<_>` return type)

Reply:
I'll try it out  - it also crossed my mind, but because it's called from `to_cookie_string` which takes '&self` I didn't think it was possible. I may learn new things about ownership though...